# Newton-Raphson Method

In [1]:
import numpy as np
import pandas as pd
import statsmodels.api as sm

# Sample data
data = {
    'Travel_Time': [30, 45, 20, 50],
    'Cost': [2.5, 3.0, 2.0, 4.0],
    'Income': [40000, 50000, 60000, 55000],
    'Age': [25, 30, 35, 28],
    'Gender': [1, 0, 1, 0],
    'Mode_Choice': [0, 1, 0, 2]  # 0 = Car, 1 = Bus, 2 = Taxi
}

# Convert data to a DataFrame
df = pd.DataFrame(data)

# Create dummy variables for the Mode_Choice column
df['Car'] = np.where(df['Mode_Choice'] == 0, 1, 0)
df['Bus'] = np.where(df['Mode_Choice'] == 1, 1, 0)
df['Taxi'] = np.where(df['Mode_Choice'] == 2, 1, 0)

# Define the independent variables (X) and dependent variables (y)
X = df[['Travel_Time', 'Cost', 'Income', 'Age', 'Gender']]
y_car = df['Car']
y_bus = df['Bus']
y_taxi = df['Taxi']

# Add a constant to the independent variables
X = sm.add_constant(X)

# Fit the Probit model for each mode
probit_model_car = sm.Probit(y_car, X).fit()
probit_model_bus = sm.Probit(y_bus, X).fit()
probit_model_taxi = sm.Probit(y_taxi, X).fit()

# Print the summary of each model
print(probit_model_car.summary())
print(probit_model_bus.summary())
print(probit_model_taxi.summary())



         Current function value: 0.000029
         Iterations: 35
         Current function value: 0.000000
         Iterations: 35
         Current function value: 0.000000
         Iterations: 35
                          Probit Regression Results                           
Dep. Variable:                    Car   No. Observations:                    4
Model:                         Probit   Df Residuals:                        0
Method:                           MLE   Df Model:                            3
Date:                Wed, 01 Jan 2025   Pseudo R-squ.:                   1.000
Time:                        19:41:05   Log-Likelihood:            -0.00011775
converged:                      False   LL-Null:                       -2.7726
Covariance Type:            nonrobust   LLR p-value:                    0.1360
                  coef    std err          z      P>|z|      [0.025      0.975]
-------------------------------------------------------------------------------
const     

C:\ProgramData\anaconda3\Lib\site-packages\statsmodels\discrete\discrete_model.py:227: PerfectSeparationWarning: Perfect separation or prediction detected, parameter may not be identified
  warnings.warn(msg, category=PerfectSeparationWarning)
C:\ProgramData\anaconda3\Lib\site-packages\statsmodels\discrete\discrete_model.py:227: PerfectSeparationWarning: Perfect separation or prediction detected, parameter may not be identified
  warnings.warn(msg, category=PerfectSeparationWarning)
C:\ProgramData\anaconda3\Lib\site-packages\statsmodels\discrete\discrete_model.py:227: PerfectSeparationWarning: Perfect separation or prediction detected, parameter may not be identified
  warnings.warn(msg, category=PerfectSeparationWarning)
C:\ProgramData\anaconda3\Lib\site-packages\statsmodels\discrete\discrete_model.py:227: PerfectSeparationWarning: Perfect separation or prediction detected, parameter may not be identified
  warnings.warn(msg, category=PerfectSeparationWarning)
C:\ProgramData\anaconda3

# Key elements in the summary:

## Coefficient (coef): Estimated effect of each independent variable on the probability of choosing the mode.

## Standard Error (std err): Measure of the precision of the coefficient estimates.

## Z-value (z): Test statistic for the coefficient.

## P-value (P>|z|): Probability that the coefficient is different from zero (statistical significance).

## Confidence Interval ([0.025, 0.975]): Range within which the true coefficient value lies with 95% confidence.